builder/dockerfile/containerbackend.go

```diff
@@ -0,0 +1,156 @@
+package dockerfile
```

Comment:
There are still a lot of builder specific pieces in here. Logging is a big one.

I understand you don't consider this part of the builder, but I don't think adding to daemon is appropriate either. I see this as part of the adapter in this model.

```

------------      --------------------      --------------
|  builder |  --> |  builder adaptor | ---> | image API |
------------      --------------------  |   --------------
                                        |   -----------------
                                        `-> | Container API |
                                            -----------------
```

(from https://github.com/moby/moby/issues/32904#issuecomment-297870519)

How about I move it out of `dockerfile` into something like `builder/containeradaptor` ?